lib/pbench/agent/tool_meister.py

```diff
@@ -844,32 +881,37 @@ def __enter__(self):
         version, seqno, sha1, hostdata = collect_local_info(self.pbench_install_dir)
 
         tool_installs = {}
-        for name, tool_opts in sorted(self._tools.items()):
+        for name, tool_opts in sorted(self._params.tools.items()):
             tklass = self._tool_name_class_mappings.get(name, TransientTool)
             try:
                 tool = tklass(
                     name,
                     tool_opts,
                     pbench_install_dir=self.pbench_install_dir,
-                    tool_dir=self._tool_dir,
                     logger=self.logger,
                 )
-                # FIXME - consider running these in parallel.
-                tool_installs[name] = tool.install()
-            except Exception:
+                # Note - each ".install()" method runs sequentially.
```

Comment:
Technically, we shouldn't be attempting to document here the *internal* behavior of the method we're calling. Rather, we might comment (probably on the outside of the loop for better context) that we run the selected `tool.install()` methods sequentially. That is, "each `.install`" implies that you're talking about the behavior of a single call.

But that's semantic nit I just can't help pointing out, not a big deal.

Reply:
How is this internal to the method?

We have a sequential loop which does not use co-routines or threads itself, and invokes a method and only invokes the next method when the current method call returns.

There are no other method calls defined to be used.

So this does not appear documenting internal behavior, but reminding the reading that this is the expected externally observed behavior of objects of the Tool class hierarchy.

I am happy to remove the comment and file an issue to track work to remove that behavior in the future instead.

Reply:
Sorry; semantic subtlety here. "each method runs sequentially" implies independent individual behavior rather than collective behavior. Therefore "they run sequentially" rather than "each runs sequentially". Sigh; I'm feeling guilty here, so nevermind.

Reply:
I've removed the comment and filed an issue: https://github.com/distributed-system-analysis/pbench/issues/2721.